File: bench/node/matmul.js
const N = parseInt(process.env.BENCH_N || "175", 10);
const MOD = 1000000007;

const A = [], B = [];
for (let i = 0; i < N; i++) {
  const ra = [], rb = [];
  for (let j = 0; j < N; j++) {
    ra.push((i + j) % 100);
    rb.push((i * j) % 100);
  }
  A.push(ra);
  B.push(rb);
}

let total = 0;
for (let i = 0; i < N; i++) {
  const Ai = A[i];
  for (let j = 0; j < N; j++) {
    let s = 0;
    for (let k = 0; k < N; k++) s += Ai[k] * B[k][j];
    total += s;
  }
}

console.log(total % MOD);

File: bench/ruby/matmul.rb
n = (ENV["BENCH_N"] || "175").to_i
mod = 1000000007

a = Array.new(n) { |i| Array.new(n) { |j| (i + j) % 100 } }
b = Array.new(n) { |i| Array.new(n) { |j| (i * j) % 100 } }

total = 0
(0...n).each do |i|
  ai = a[i]
  (0...n).each do |j|
    s = 0
    (0...n).each { |k| s += ai[k] * b[k][j] }
    total += s
  end
end

puts total % mod
